app/controllers/email_signups_controller.rb

```diff
@@ -1,32 +1,38 @@
 class EmailSignupsController < PublicFacingController
   def new
-    feed_url = email_signup_params['feed']
-
-    if feed_url.match?(%r{/government/organisations/(.*)\.atom$})
-      redirect_to email_alert_frontend_signup(feed_url)
+    if non_finder_url
+      redirect_to email_alert_frontend_signup
+    elsif feed_url.match?(%r{/government/publications\.atom(.*)$})
+      redirect_to feed_url.sub('.atom', '')
+    elsif feed_url.match?(%r{/government/announcements\.atom(.*)$})
+      redirect_to feed_url.sub('.atom', '')
+    elsif feed_url.match?(%r{/government/statistics\.atom(.*)$})
+      redirect_to feed_url.sub('.atom', '')
     else
-      @email_signup = EmailSignup.new(email_signup_params)
+      redirect_to '/'
     end
   end
 
-  def create
-    @email_signup = EmailSignup.new(email_signup_params)
+private
 
-    if @email_signup.save
-      redirect_to @email_signup.signup_url
-    else
-      render action: 'new'
-    end
+  def non_finder_url
+    feed_url.match?(%r{/government/organisations/(.*)\.atom$}) ||
+      feed_url.match?(%r{/government/ministers/(.*)\.atom$}) ||
+      feed_url.match?(%r{/government/people/(.*)\.atom$}) ||
+      feed_url.match?(%r{/government/topical-events/(.*)\.atom$}) ||
+      feed_url.match?(%r{/world/(.*)\.atom$})
   end
 
-private
-
   def email_signup_params
     params.require(:email_signup).permit(:feed)
```

Comment:
might want to require feed, eg `params.require(:email_signup).require(:feed)`. Then we can probably avoid the feed_url method.

Reply:
Sure